### PR TITLE
Null check / Typos / Upgraded assemblies / Fixed connection events

### PR DESCRIPTION
**Pull request by fozzzgate**

### CODE REVIEW

**Comment by flyingxu** [comment]
Thanks, I will be on travel the next week so I will merge it for now and look into it later.
As you may also have seen, this repo is a fork from `xabre/xamarin-bluetooth-le`. To make your contribution visible to more users, I would like to suggest you also create a PR to that repo as well. There is a `uwp_creators_update` branch in that repo and the code is mostly working. I compared that branch with my code but I found many differences in coding so I'm a little hesitant to create a PR of my code to that branch. How do you think?

**Comment by fozzzgate** [comment]
I submitted to PR to you because you seem to be the main developer on the UWP fork. But I can submit the changes to the Xabre repository as well.

I think the coding style is fine. Saved me tons of time. Whether or not the code works is the important thing and this one works great. Compared to a different Bluetooth LE library that I was previously using, this one is much less buggy and more reliable.

File: Source/Plugin.BLE.UWP/Plugin.BLE.UWP.csproj [hunk 1]
```diff
@@ -124,8 +124,8 @@
     <PackageReference Include="Microsoft.NETCore.UniversalWindowsPlatform">
       <Version>5.4.0</Version>
     </PackageReference>
-    <PackageReference Include="Microsoft.Toolkit.Uwp">
-      <Version>1.5.0</Version>
+    <PackageReference Include="Microsoft.Toolkit.Uwp.Connectivity">
```
Comment (flyingxu) [inline Source/Plugin.BLE.UWP/Plugin.BLE.UWP.csproj]: Please make sure this change get PRed to the Xabre repo.

**Comment by flyingxu** [comment]
Thanks again.
I'm active in the UWP platform implementation because I need to release a UWP Bluetooth app in the coming months which will have several connections simultaneously. I'm not the main developer of the UWP branch, my code should still be reviewed and approved before getting merged, just like any other one who is willing to spend time in open source project.
Please go ahead and get your code PRed to Xabre repo which will also be my main working branch.

**Comment by fozzzgate** [comment]
I just submitted my PRs to the Xabre base repository. When you get a chance, please submit a PR to them for your Scan Extensions so that all the code is synchronized.

**Comment by fozzzgate** [comment]
Xabre accepted my PRs

**Pull request merged**